src/server/scripts/Kalimdor/boss_azuregos.cpp

```diff
@@ -15,156 +15,220 @@
+                    DoMeleeAttackIfReady();
+                });
+        }
 
-            //ChillTimer
-            if (ChillTimer <= diff)
-            {
-                DoCastVictim(SPELL_CHILL);
-                ChillTimer = urand(13000, 25000);
-            }
-            else ChillTimer -= diff;
+        protected:
+            TaskScheduler _scheduler;
+    };
 
-            //BreathTimer
-            if (BreathTimer <= diff)
-            {
-                DoCastVictim(SPELL_FROSTBREATH);
-                BreathTimer = urand(10000, 15000);
-            }
-            else BreathTimer -= diff;
+    bool OnGossipSelect(Player* player, Creature* creature, uint32 /*sender*/, uint32 /*action*/) override
+    {
+        CloseGossipMenuFor(player);
+        creature->SetFaction(FACTION_ENEMY);
+        creature->AI()->AttackStart(player);
+        return true;
+    }
 
-            //ManaStormTimer
-            if (ManaStormTimer <= diff)
-            {
-                if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
-                    DoCast(target, SPELL_MANASTORM);
-                ManaStormTimer = urand(7500, 12500);
-            }
-            else ManaStormTimer -= diff;
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new boss_azuregosAI(creature);
+    }
+};
 
-            //ReflectTimer
-            if (ReflectTimer <= diff)
-            {
-                DoCast(me, SPELL_REFLECT);
-                ReflectTimer = urand(20000, 35000);
-            }
-            else ReflectTimer -= diff;
+// Arcane Vacuum: 21147
+class spell_arcane_vacuum : public SpellScript
+{
+    PrepareSpellScript(spell_arcane_vacuum);
 
-            //CleaveTimer
-            if (CleaveTimer <= diff)
-            {
-                DoCastVictim(SPELL_CLEAVE);
-                CleaveTimer = 7000;
-            }
-            else CleaveTimer -= diff;
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_ARCANE_VACUUM, SPELL_ARCANE_VACUUM_TP });
+    }
 
-            //EnrageTimer
-            if (HealthBelowPct(26) && !Enraged)
-            {
-                DoCast(me, SPELL_ENRAGE);
-                Enraged = true;
-            }
+    void HandleOnHit()
+    {
+        Unit* caster = GetCaster();
+        Unit* hitUnit = GetHitUnit();
+        if (caster && hitUnit && hitUnit->ToPlayer())
+        {
+            caster->CastSpell(hitUnit, SPELL_ARCANE_VACUUM_TP, true);
+        }
+    }
+
+    void Register() override
+    {
+        OnHit += SpellHitFn(spell_arcane_vacuum::HandleOnHit);
+    }
+};
+
+// Mark of Frost - Triggered Spell
+class spell_mark_of_frost_freeze : public SpellScript
+{
+    PrepareSpellScript(spell_mark_of_frost_freeze);
 
-            DoMeleeAttackIfReady();
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_MARK_OF_FROST, SPELL_AURA_OF_FROST, SPELL_MARK_OF_FROST_AURA });
```

Comment:
```suggestion
        return ValidateSpellInfo({ SPELL_MARK_OF_FROST, SPELL_AURA_OF_FROST });
```
Only validate spells you use inside the script